framework/src/base/ResidualObject.C

```diff
@@ -62,3 +54,9 @@ ResidualObject::ResidualObject(const InputParameters & parameters, bool is_nodal
     _mesh(_subproblem.mesh())
 {
 }
+
+void
+ResidualObject::prepareShapes(const unsigned int var_num)
+{
+  _subproblem.prepareShapes(var_num, _tid);
```

Comment:
What the prepare shapes is doing? Are the two kernels having the same thing thus we could be duplicating computation here?